backend/connectors/rasa-connector/src/main/java/co/airy/core/rasa_connector/RasaConnectorService.java

```diff
@@ -0,0 +1,67 @@
+package co.airy.core.rasa_connector;
+
+import co.airy.avro.communication.Message;
+import co.airy.core.rasa_connector.models.MessageSend;
+import co.airy.core.rasa_connector.models.MessageSendResponse;
+import co.airy.log.AiryLoggerFactory;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class RasaConnectorService {
+    //private final apiToken;
+    private final RasaClient rasaClient;
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private static final Logger log = AiryLoggerFactory.getLogger(RasaConnectorService.class);
+    private final MessageHandler messageHandler;
+    RasaConnectorService(MessageHandler messageHandler,
+                         RasaClient rasaClient){
+        this.rasaClient = rasaClient;
+        this.messageHandler = messageHandler;
+    }
+
+    @Async("threadPoolTaskExecutor")
+    public void send(Message userMessage) {
+        try {
+            List<MessageSendResponse> rasaResponseList = this.rasaClient.sendMessage(MessageSend.builder()
+                    .message(getTextFromContent(userMessage.getContent()))
+                    .sender(userMessage.getId())
+                    .build());
+            // Unpack multiple response(s)
+            for (MessageSendResponse rasaResponse: rasaResponseList) {
+                try {
+                    messageHandler.writeReplyToKafka(userMessage, rasaResponse);
```

Comment:
@armanjindal this is good, but doing it [this wasy](https://github.com/airyhq/airy/pull/3614/files#diff-3f2f43c5b7b114c93a58727cc16cddb743f7e498250bc5aef09dc4b616fc7aabR117) is better because it gives us better guarantees: Processing will be retried properly when writing to Kafka fails.

Reply:
Thanks for the tip @chrismatix! The link points me to line 117 of the `EventsRouter.java` file and the `.to` method there. Is this what you meant to link to? Are you perhaps referencing `MessageMapper.java` file? Since it seems like you do add the responses to a list instead of sending one at a time. I'll keep digging

Reply:
@armanjindal Yes, so how this works is that this `send` method returns a `List<KeyValue<String, Record>>` where the pairs are what you would send to Kafka manually. In the `to` method you then send records to topics depending on type.